src/metadata_parser/__init__.py

```diff
@@ -313,9 +312,11 @@ def get_encoding_from_headers(headers: CaseInsensitiveDict) -> Optional[str]:
     content_type = headers.get("content-type")
     if not content_type:
         return None
-    content_type, params = cgi.parse_header(content_type)
-    if "charset" in params:
-        return params["charset"].strip("'\"")
+    if not "charset" in content_type:
+        return None
+    for param in content_type.replace(" ", "").split(";"):
+        if "charset=" in param:
+            return param.split("=")[-1]
```

Comment:
this should be (or similar):

    return param.split("=")[-1].strip("'\"")
    
content-type can often look like either: the rfc allows for quoted values, not just "tokens".

    content-type: charset=utf8
    content-type: charset="utf8"
   
reference: https://datatracker.ietf.org/doc/html/rfc5987#section-3.2.2

Reply:
But in this case 
```
return param.split("=")[-1].strip("'\"")
``` 

If we are `text/html;charset=utf8`, value return with script is *
```
[
"text/html;charset",
"utf8"
]
```

no ?